# HG changeset patch
# User Mike Miller <mtmiller@ieee.org>
# Date 1384484978 18000
#      Thu Nov 14 22:09:38 2013 -0500
# Node ID a404853d20734714c9299699ac4cdf8151b34d5a
# Parent  ffdbb82a0c78dfd625d4621aaa90aa13e3958714
qp.m: Condition test on HAVE_GLPK

* qp.m: Condition test from changeset 185038fe7a16 on HAVE_GLPK.

diff --git a/scripts/optimization/qp.m b/scripts/optimization/qp.m
--- a/scripts/optimization/qp.m
+++ b/scripts/optimization/qp.m
@@ -403,17 +403,18 @@ function [x, obj, INFO, lambda] = qp (x0
 
   else
     print_usage ();
   endif
 
 endfunction
 
 
-%!test  # with infeasible initial guess
+%% Test infeasible initial guess (bug #40536)
+%!testif HAVE_GLPK
 %!
 %! H = 1;  q = 0;                # objective: x -> 0.5 x^2
 %! A = 1;  lb = 1;  ub = +inf;   # constraint: x >= 1
 %! x0 = 0;                       # infeasible initial guess
 %!
 %! [x, obj_qp, INFO, lambda] = qp (x0, H, q, [], [], [], [], lb, A, ub);
 %!
 %! assert (isstruct (INFO) && isfield (INFO, "info") && (INFO.info == 0));
